### PR TITLE
Update actions to v3

### PR DESCRIPTION
**Pull request by leemyongpakvn**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Update actions cache v2, upload artifact v1, download artifact v1 to v3 to solve deprecated warnings.
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #{issue URL here}, Fixes #{another issue URL here}
| Sponsor company   | Your company or customer's name goes here (if applicable).
| How to test?      | Build Release green in modules that use this common yml

### CODE REVIEW

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -18,17 +18,17 @@ jobs:
       - name: Generate index.php
         run: ~/.composer/vendor/bin/autoindex
       - name: Create & upload artifact
-        uses: actions/upload-artifact@v1
+        uses: actions/upload-artifact@v3
         with:
           name: ${{ github.event.repository.name }}
-          path: ../
+          path: /home/runner/work/${{ github.event.repository.name }}
```
Comment (kpodemski) [inline .github/workflows/build-release.yml:24]: is it necessary to change this? do you have any source confirming this change?

Comment (kpodemski) [reply .github/workflows/build-release.yml:24]: @leemyongpakvn

Comment (leemyongpakvn) [reply .github/workflows/build-release.yml:24]: @kpodemski upload-artifact v3 does not allow ../ as path, I will change back to ../ then you will can the error.
The first confirmation is at https://github.com/PrestaShop/blockwishlist/pull/216/files
The second confirmation is at https://github.com/PrestaShop/pagesnotfound/pull/29/files :)

**Review by kpodemski — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @leemyongpakvn

**Pull request merged**